core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -499,6 +499,25 @@ class SnappyConf(@transient val session: SnappySession)
           pool.toString
         case Some(pool) => throw new IllegalArgumentException(s"Invalid Pool $pool")
       }
+
+    case Property.PlanCaching.name =>
+      value match {
+        case Some(boolval) => {
+          if (boolval.toString.toBoolean) session.clearPlanCache()
+        }
+        case None =>
+      }
+
+    case Property.PlanCachingAll.name =>
+      value match {
+        case Some(boolval) => {
+          val disable_tokenization = !boolval.toString.toBoolean
+          System.setProperty("DISABLE_TOKENIZATION", disable_tokenization.toString)
+          if (disable_tokenization) SnappySession.getPlanCache.asMap().clear()
+        }
```

Comment:
Why mixing tokenization with planCaching flag? The two are separate things. I think better to make tokenization a separate property (e.g. snappydata.sql.tokenization=true/false) like mentioned in previous comment.

Reply:
Ok. will add the property looks good.
BTW We do tokenize to see if we cache plan or not. So won't they go together. So did not quite understand the comment - "Why these two things are being mixed"

Reply:
Tokenization has significant value even if plan caching is disabled. Because its major benefit lies in being able to produce same generated code for two different logical plans that differ only in the embedded constants. Plan caching is primarily for low latency queries (< 50ms or so), while tokenization helps all kinds of queries (e.g. we have seen first runs of queries that require compilation+JIT to be many times slower, or the TPCH runs with different values where most of the gains are due to tokenization).

Reply:
Actually I don't understand this part well. I mean generated code is part of the plan which is getting cached or not? So plan caching before Tokenization only helped when the same query ( with the same constants ) ran again and again. If a new query with just different constants made the engine produce new plan and hence new generated code. So yes they are different but if different queries are being run ( even differeing in constants ) then doesn't plan caching and Tokenization in some sense means the same thing.

Reply:
Plan caching is different from code generation cache. Latter is common across plans/sessions which is where the biggest advantage of tokenization comes.

Plan caching reduces the work to recreate LogicalPlan=>optimize=>...=>physical  SparkPlan generation. This is normally only a factor for concurrent short queries (in the range of few ms). The physical plan then generates code which is cached in Spark's CodeGenerator.cache that does exact generated code string lookup. That is where tokenization allows to generate same code.

> So a new query with just different constants made the engine produce new plan and hence new generated code.

Plan cache is for a session. Why would same query or query patterns in different sessions produce different generated code string? That would a critical bug. I had fixed one issue in ColumnTableScan MIN/MAX predicate code where code string would be different due to predicate ordering being unpredictable due to hash map iteration (so had changed to sorted consistent ordering). We need to ensure that generated code across sessions for same query patterns (i.e. differing only in constants) should be identical.

Reply:
@sumwale, just curious to know if we can have cases where planCache will be a cache hit but for same query code generation cache will be a cache miss. I mean if the sparkPlan/executedPlan is available in planCache will there always be corresponding generated code in Code generation cache. Or size/capacity of these caches will also play a role here. planCache can hold 3000 plans but default value for spark.sql.codegen.cacheSize is 250

Reply:
That's possible. In snappy-spark the Spark codegen cache is 1000